vault/resource_kmip_secret_backend.go

```diff
@@ -0,0 +1,222 @@
+			},
+
+			"tls_ca_key_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "CA key type, rsa or ec",
+			},
+			"tls_ca_key_bits": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "CA key bits, valid values depend on key type",
+			},
+			"tls_min_version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Minimum TLS version to accept",
+			},
+			"default_tls_client_key_type": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Client certificate key type, rsa or ec",
+			},
+			"default_tls_client_key_bits": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "Client certificate key bits, valid values depend on key type",
+			},
+			"default_tls_client_ttl": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "Client certificate TTL in seconds",
+			},
+		},
+	}
+}
+
+func kmipSecretBackendCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	path := d.Get("path").(string)
+	description := d.Get("description").(string)
+	defaultTLSClientTTL := d.Get("default_tls_client_ttl").(int)
+
+	log.Printf("[DEBUG] Mounting KMIP backend at %q", path)
+	if err := client.Sys().Mount(path, &api.MountInput{
+		Type:        "kmip",
+		Description: description,
+		Config: api.MountConfigInput{
+			DefaultLeaseTTL: fmt.Sprintf("%d", defaultTLSClientTTL),
+		},
+	}); err != nil {
+		return fmt.Errorf("error mounting to %q: %s", path, err)
+	}
+
+	log.Printf("[DEBUG] Mounted KMIP backend at %q", path)
+	d.SetId(path)
+
+	return kmipSecretBackendUpdate(d, meta)
+}
+
+func kmipSecretBackendUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	path := d.Id()
+	tune := api.MountConfigInput{}
+
+	if d.HasChange("default_tls_client_ttl") || d.HasChange("description") {
+		tune.DefaultLeaseTTL = fmt.Sprintf("%ds", d.Get("default_tls_client_ttl"))
+		description := d.Get("description").(string)
+		tune.Description = &description
+
+		log.Printf("[DEBUG] Updating mount for %q", path)
+		err := client.Sys().TuneMount(path, tune)
+		if err != nil {
+			return fmt.Errorf("error updating mount for %q, err=%w", path, err)
+		}
+		log.Printf("[DEBUG] Updated mount for %q", path)
+	}
+
+	data := map[string]interface{}{}
+	configPath := fmt.Sprintf("%s/config", path)
+	log.Printf("[DEBUG] Updating %q", configPath)
+
+	for _, k := range kmipAPIFields {
+		if d.HasChange(k) {
+			if v, ok := d.GetOk(k); ok {
+				switch v.(type) {
+				case *schema.Set:
+					data[k] = util.TerraformSetToStringArray(v)
+				default:
+					data[k] = v
+				}
+			}
+		}
+	}
+
+	if _, err := client.Logical().Write(configPath, data); err != nil {
```

Comment:
Is there any particular reason we don't do this before tuning the mount? Is the order of operation important here?

Reply:
I'm not 100% whether the order of operns matters, but I followed the precedent laid in the `Create` method, which deals with setting up the mount first and then writes the data.